From: Miao Xie <miaox@cn.fujitsu.com>
Date: Wed, 8 Aug 2012 21:39:36 -0600
Patch-mainline: 3.6
Git-commit: c0f62dedd04ae0f3b8a18079db5a015af24e416f
References: FATE#312888
Subject: [PATCH] Btrfs: fix wrong mtime and ctime when creating
 snapshots

When we created a new snapshot, the mtime and ctime of its parent directory
were not updated. Fix it.

Signed-off-by: Miao Xie <miaox@cn.fujitsu.com>
Signed-off-by: Chris Mason <chris.mason@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/transaction.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/btrfs/transaction.c
+++ b/fs/btrfs/transaction.c
@@ -1024,6 +1024,7 @@ static noinline int create_pending_snaps
 
 	btrfs_i_size_write(parent_inode, parent_inode->i_size +
 					 dentry->d_name.len * 2);
+	parent_inode->i_mtime = parent_inode->i_ctime = CURRENT_TIME;
 	ret = btrfs_update_inode(trans, parent_root, parent_inode);
 	if (ret)
 		goto abort_trans_dput;
